### PR TITLE
fix(PatternBrush): `getPatternSrc`, rm `getPatternSrcFunction`

### PR DESCRIPTION
**Pull request by ShaMan123**
## Motivation

https://github.com/fabricjs/fabric.js/pull/8455#discussion_r1032098929

## Description

`getPatternSrcFunction` was weird, mutating the function itself by converting it to string. Bad practice and no need for it.
It seemed that in v4/v5 Pattern stopped accepting strings to execute as functions, breaking PatternBrush. It was forgotten since (I am guessing) it had no tests.
So I have removed the method to align with Pattern and added tests that cover these cases.
Refactored `PatternBrush.getPatternSrc` signature to accept color as an arg. Am not sure it is needed to say the truth. We can use `this.color` because the function runs in context always, seems so.

## Changes

Non breaking since `this.color` will still work as expected in `getPatternSrc`

## Gist

Added visual tests.
Found  a leak in test config.

### CODE REVIEW

File: test/visual/freedraw.js [hunk 1]
```diff
@@ -2056,9 +2015,43 @@
...
+    else {
+      fabric.config.configure({ browserShadowBlurConstant: BROWSER_SHADOW_BLUR[getBrowserAgent()] });
+    }
+    fabric.config.configure({
+      enableGLFiltering: false
+    });
+    fabric.Object.prototype.objectCaching = true;
+  });
+  hooks.after(() => {
+    // fabric.config.restoreDefaults();
```
Comment (ShaMan123) [inline test/visual/freedraw.js:2048]: This is really bad.
Other suites depend on some configuration made in this test that leaks.

**Review by ShaMan123 — COMMENTED** [review]
READY and TESTED

**Comment by asturur** [comment]
The patternBrush is more likely a demo / blueprint rather than something useful.
I don't think the color parameter is needed, all other brushes work using the color property on the instance

**Comment by asturur** [comment]
on top of that getPatternSrcFunction imho was a possible security flaw.
Imaging adding to this.color a kind of function that then gets executed with the same rights as any javascript loaded on the website.
Definetely a security issue, reason why Pattern does not have that function anymore

**Review by asturur — APPROVED** [review]

**Pull request merged**